src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.name = name;
+  }
+
+  bite(beast) {
+    if (beast.hidden === false && beast instanceof Carnivore === false) {
+      beast.health -= 50;
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
You can firstly check `if (beast.health === 0)` only after that call the filter method. This is necessary in order to prevent unnecessary calls to the filter method